### PR TITLE
IBM Security Verify Collector - ciac 8132

### PR DESCRIPTION
**Pull request by YairGlik**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-8132)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by shmuel44 — COMMENTED** [review]
Approved from my side. (native image)

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/IBMSecurityVerifyEventCollector_description.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+The integration is designed for customers using IBM Security Verify to collect and analyze security events across their organization's network. By connecting to the IBM Security Verify Event Collector, organizations can effectively monitor, manage, and respond to potential security incidents in real-time.
+
+The API Credentials include a combination of **Client ID** and **Client Secret**. These credentials are equivalent to a username and password and should be handled with care. It is strongly recommended to keep the **Client ID** and **Client Secret** secure and not share them with unauthorized parties to prevent potential security breaches.
+
+To obtain the **Client ID** and **Client Secret**, follow these steps:
+
+1. **Log in to the IBM Security Verify UI**.
+2. **Click on the profile icon** located at the top right corner of the interface.
+3. **Select "Switch to admin"** to access administrative settings.
```
Comment (ShirleyDenkberg) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/IBMSecurityVerifyEventCollector_description.md:11]:
```suggestion
1. Log in to the IBM Security Verify UI.
2. Click the profile icon located at the top right corner of the interface.
3. Select **Switch to admin** to access administrative settings.
```

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/IBMSecurityVerifyEventCollector_description.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+4. Navigate to **Security > API Access**.
```
Comment (ShirleyDenkberg) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/IBMSecurityVerifyEventCollector_description.md:12]:
```suggestion
4. Navigate to **Security** > **API Access**.
```

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/IBMSecurityVerifyEventCollector_description.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+5. Click on **Add API Client** to generate the necessary credentials.
```
Comment (ShirleyDenkberg) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/IBMSecurityVerifyEventCollector_description.md:13]:
```suggestion
5. Click **Add API Client** to generate the necessary credentials.
```

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/IBMSecurityVerifyEventCollector_description.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+6. After clicking on **Add API Client**, make sure to assign the following permissions to the API client:
```
Comment (ShirleyDenkberg) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/IBMSecurityVerifyEventCollector_description.md:14]:
```suggestion
6. After clicking **Add API Client**, make sure to assign the following permissions to the API client:
```

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/IBMSecurityVerifyEventCollector_description.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+## IBM Security Verify Event Collector
+
+The integration is designed for customers using IBM Security Verify to collect and analyze security events across their organization's network. By connecting to the IBM Security Verify Event Collector, organizations can effectively monitor, manage, and respond to potential security incidents in real-time.
+
+The API Credentials include a combination of **Client ID** and **Client Secret**. These credentials are equivalent to a username and password and should be handled with care. It is strongly recommended to keep the **Client ID** and **Client Secret** secure and not share them with unauthorized parties to prevent potential security breaches.
+
```
Comment (ShirleyDenkberg) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/IBMSecurityVerifyEventCollector_description.md:6]:
```suggestion

Follow these steps to ensure that you are in the correct administrative mode before accessing the API credentials, which are essential for connecting and interacting with the IBM Security Verify Event Collector. 

```

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/README.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
+IBM Security Verify provides a secure and scalable solution for collecting and managing security events from IBM Security Verify, offering advanced threat detection and response capabilities for protecting identities, applications, and data.
+
+## Set up the Third Party System
+To obtain the **Client ID** and **Client Secret**, follow these steps:
+
+1. **Log in to the IBM Security Verify UI**.
+2. **Click on the profile icon** located at the top right corner of the interface.
+3. **Select "Switch to admin"** to access administrative settings.
```
Comment (ShirleyDenkberg) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/README.md:8]:
```suggestion
1. Log in to the IBM Security Verify UI.
2. Click the profile icon located at the top right corner of the interface.
3. Select **Switch to admin** to access administrative settings.
```

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/README.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+4. Navigate to **Security > API Access**.
+5. Click on **Add API Client** to generate the necessary credentials.
+6. After clicking on **Add API Client**, make sure to assign the following permissions to the API client:
```
Comment (ShirleyDenkberg) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/README.md:11]:
```suggestion
5. Click **Add API Client** to generate the necessary credentials.
6. After clicking **Add API Client**, make sure to assign the following permissions to the API client:
```

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/README.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+   - **Manage reports**
+   - **Read reports**
+
+- ![Creating an API Client](../../doc_files/Creating_an_API_Client.gif)
+
+## Configure IBM Security Verify Event Collector on Cortex XSOAR
```
Comment (ShirleyDenkberg) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/README.md:17]:
```suggestion
## Configure IBM Security Verify Event Collector on Cortex XSIAM
```

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/README.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+1. Navigate to **Settings** > **Integrations** > **Servers & Services**.
```
Comment (ShirleyDenkberg) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/README.md:19]:
```suggestion
1. Navigate to **Settings** > **Configurations** > **Data Collection** > **Automations & Feed Integrations**.
```

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/README.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    | Client Secret |  | True |
+    | The maximum number of events per fetch | The maximum is 50,000. | True |
+    | Trust any certificate (not secure) |  | False |
+    | Use system proxy settings |  | False |
+
+4. Click **Test** to validate the URLs, token, and connection.
+
+## Commands
+
+You can execute these commands from the Cortex XSOAR CLI, as part of an automation, or in a playbook.
```
Comment (ShirleyDenkberg) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerifyEventCollector/README.md:36]:
```suggestion
You can execute these commands from the Cortex XSIAM CLI, as part of an automation, or in a playbook.
```

File: Packs/IBMSecurityVerify/README.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
+## Secure and Manage Your Identity Data with IBM Security Verify
+
+Identity and access management is a crucial part of maintaining security in modern enterprises. IBM Security Verify provides robust solutions to monitor, control, and secure user access to critical systems. This integration helps automate the collection and analysis of security events, ensuring a streamlined and secure identity management process.
+
+**IBM Security Verify Integration Highlights:**
+
+- Centralized Monitoring: Collect security events from the IBM Security Verify EventCollector for centralized monitoring and analysis.
+- Enhanced Security: Use the power of Cortex XSOAR to respond to security incidents involving identity and access management.
```
Comment (ShirleyDenkberg) [inline Packs/IBMSecurityVerify/README.md:8]:
```suggestion
- Enhanced Security: Use the power of Cortex XSIAM to respond to security incidents involving identity and access management.
```

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/README.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+3. **Select "Switch to admin"** to access administrative settings.
+4. Navigate to **Security > API Access**.
+5. Click on **Add API Client** to generate the necessary credentials.
+6. After clicking on **Add API Client**, make sure to assign the following permissions to the API client:
+   - **Manage reports**
+   - **Read reports**
+
+- ![Creating an API Client](../../doc_files/Creating_an_API_Client.gif)
+
+## Configure IBM Security Verify Event Collector on Cortex XSOAR
```
Comment (ShirleyDenkberg) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/README.md:17]:
```suggestion
## Configure IBM Security Verify Event Collector on Cortex XSIAM
```

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/README.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+    | Client Secret |  | True |
+    | The maximum number of events per fetch | The maximum is 50,000. | True |
+    | Trust any certificate (not secure) |  | False |
+    | Use system proxy settings |  | False |
+
+4. Click **Test** to validate the URLs, token, and connection.
+
+## Commands
+
+You can execute these commands from the Cortex XSOAR CLI, as part of an automation, or in a playbook.
```
Comment (ShirleyDenkberg) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/README.md:36]:
```suggestion
You can execute these commands from the Cortex XSIAM CLI, as part of an automation, or in a playbook.
```

**Comment by ShirleyDenkberg** [comment]
@shmuel44 @jbabazadeh Doc review completed.

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/IBMSecurityVerify.py [hunk 14]
```diff
@@ -0,0 +1,302 @@
...
+
+
+def get_events_command(client: Client, args: dict) -> tuple[list[dict], str]:
+    """
+    Retrieves events using the client based on the provided arguments.
+    """
+
+    last_id = args.get("last_id")
+    last_time = args.get("last_time")
+    last_item = {"last_id": last_id, "last_time": last_time}
```
Comment (jbabazadeh) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/IBMSecurityVerify.py:132]: add check that if provided last_id or last_time both of them should be provided.

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/IBMSecurityVerify.yml [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+      auto: PREDEFINED
+      isArray: false
+      predefined:
+      - 'True'
+      - 'False'
+    - name: limit
+      description: Maximum number of results to return.
+      defaultValue: 1000
+    - name: last_id
+      description: "The ID of the last event retrieved. Use together with `last_time` for pagination to get events after this ID. Example: 1234abcd-5678-90ef-1234-567890abcdef."
```
Comment (jbabazadeh) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/IBMSecurityVerify.yml:58]:
```suggestion
      description: "The ID of the last event retrieved. Use together with `last_time` to get events after this ID. Example: 1234abcd-5678-90ef-1234-567890abcdef."
```

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/IBMSecurityVerify.yml [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+    - name: last_time
+      description: "The timestamp of the last event retrieved. Use together with `last_id` for pagination to get events after this time. Example: 1672531200000."
+    - name: sort_order
+      description: "Order to sort events by: 'Desc' or 'Asc'."
+      defaultValue: 'Desc'
```
Comment (jbabazadeh) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/IBMSecurityVerify.yml:63]: why this is the default value?

File: Packs/IBMSecurityVerify/doc_files/Creating_an_API_Client.gif
Comment (jbabazadeh) [inline Packs/IBMSecurityVerify/doc_files/Creating_an_API_Client.gif:1]: please remove this gif and upload it to the content_assets repo (dont you wanted to create a new gif?)

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/README.md [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+1. **Log in to the IBM Security Verify UI**.
+2. **Click on the profile icon** located at the top right corner of the interface.
+3. **Select "Switch to admin"** to access administrative settings.
+4. Navigate to **Security** > **API Access**.  
+5. Click **Add API Client** to generate the necessary credentials.
+6. After clicking **Add API Client**, make sure to assign the following permissions to the API client:
+   - **Manage reports**
+   - **Read reports**
+
+- ![Creating an API Client](../../doc_files/Creating_an_API_Client.gif)
```
Comment (jbabazadeh) [inline Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/README.md:15]: add the url from the content_assets repo

**Review by jbabazadeh — CHANGES_REQUESTED** [review]
Nice work!
please see my comments

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/IBMSecurityVerify.py [hunk 18]
```diff
@@ -0,0 +1,302 @@
...
+
+def test_module(client: Client, params) -> str:
+    """
+    'ok' if test passed, anything else will raise an exception and will fail the test.
+    """
+    args = {"limit": 1}
+    get_events_command(client, args)
+
+    if argToBoolean(params.get('isFetchEvents')):
+        max_limit_validation(arg_to_number(params.get('max_fetch')))
```
[the remark this answers is not in the document]
Comment (YairGlik) [reply Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/IBMSecurityVerify.py:121]: Great idea, thanks!

File: Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/IBMSecurityVerify.yml (hunk 16 above)
Comment (YairGlik) [reply Packs/IBMSecurityVerify/Integrations/IBMSecurityVerify/IBMSecurityVerify.yml:63]: It's a command that doesn't appear in the design, how much to write?

**Review by jbabazadeh — APPROVED** [review]

**Pull request merged**